### ISSUE TITLE
List of enums as a query argument

### ISSUE THREAD

**Issue by kono0514** [comigor/artemis#195]
**Bug description**
Build error when the query argument is a list of enums

**Specs**
Artemis version: 6.6.4-beta.1
<details>
<summary>build.yaml:</summary>
  
```yaml
targets:
  $default:
    sources:
      - lib/**
      - graphql/**
      - schema.graphql
    builders:
      artemis:
        options:
          schema_mapping:
            - schema: schema.graphql
              queries_glob: graphql/*.graphql
              output: lib/graphql_api.dart
```
</details>
<details>
<summary>Artemis output:</summary>
  
```bash
[ +166 ms] executing: [C:\flutter\flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[ +144 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[   +1 ms] 2ae34518b87dd891355ed6c6ea8cb68c4d52bb9d
[   +2 ms] executing: [C:\flutter\flutter/] git tag --contains HEAD
[ +273 ms] Exit code 0 from: git tag --contains HEAD
[   +1 ms] 1.20.1
[  +17 ms] executing: [C:\flutter\flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[  +59 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[   +1 ms] origin/stable
[        ] executing: [C:\flutter\flutter/] git ls-remote --get-url origin
[  +50 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[ +100 ms] executing: [C:\flutter\flutter/] git rev-parse --abbrev-ref HEAD
[  +50 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[   +1 ms] stable
[  +78 ms] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[  +11 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[  +17 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[  +19 ms] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[  +11 ms] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[  +11 ms] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +25 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidMavenArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[   +1 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[  +11 ms] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[  +45 ms] Using C:\flutter\flutter\.pub-cache for the pub cache.
[   +9 ms] executing: C:\flutter\flutter\bin\cache\dart-sdk\bin\pub.bat run build_runner build --verbose
[INFO] Generating build script...
[INFO] Generating build script completed, took 850ms

[WARNING] build.fallback:
The package `flutter_artemis_test` does not include some required sources in any of its targets (see their build.yaml file).
The missing sources are:
  - $package$
[INFO] BuildDefinition:Initializing inputs
[INFO] BuildDefinition:Reading cached asset graph...
[INFO] BuildDefinition:Reading cached asset graph completed, took 103ms

[INFO] BuildDefinition:Checking for updates since last build...
[INFO] BuildDefinition:Checking for updates since last build completed, took 988ms

[INFO] Build:Running build...
[INFO] Build:Running build completed, took 28ms

[INFO] Build:Caching finalized dependency graph...
[INFO] Build:Caching finalized dependency graph completed, took 87ms

[SEVERE] artemis:artemis on lib/$lib$ (cached):

Could not format because the source could not be parsed:

line 29, column 46: Expected to find '('.
   ╷
29 │ @JsonKey(unknownEnumValue: List<ArticleType>.artemisUnknown) final List<ArticleType> article_type_in;
   │                                              ^^^^^^^^^^^^^^
   ╵
package:dart_style/src/dart_formatter.dart 142:7                        DartFormatter.formatSource
package:dart_style/src/dart_formatter.dart 74:12                        DartFormatter.format
package:artemis/generator/print_helpers.dart 386:26                     specToString
package:artemis/generator/print_helpers.dart 394:16                     writeLibraryDefinitionToBuffer
package:artemis/builder.dart 148:7                                      GraphQLQueryBuilder.build
package:build                                                           runBuilder
package:build_runner_core/src/generate/build_impl.dart 485:19           _SingleBuild._runForInput.<fn>.<fn>.<fn>
package:build_runner_core/src/generate/performance_tracker.dart 302:15  _NoOpBuilderActionTracker.trackStage
package:build_runner_core/src/generate/build_impl.dart 483:23           _SingleBuild._runForInput.<fn>.<fn>
package:build_runner_core/src/generate/build_impl.dart                  _SingleBuild._runForInput.<fn>.<fn>
package:timing/src/timing.dart 222:44                                   NoOpTimeTracker.track
package:build_runner_core/src/generate/build_impl.dart 440:22           _SingleBuild._runForInput.<fn>
package:pool/pool.dart 127:28                                           Pool.withResource
package:build_runner_core/src/generate/build_impl.dart 436:17           _SingleBuild._runForInput
package:build_runner_core/src/generate/build_impl.dart 374:38           _SingleBuild._runBuilder.<fn>
dart:async                                                              Future.wait
package:build_runner_core/src/generate/build_impl.dart 373:36           _SingleBuild._runBuilder
package:build_runner_core/src/generate/build_impl.dart 319:20           _SingleBuild._runPhases.<fn>.<fn>
package:build_runner_core/src/generate/build_impl.dart                  _SingleBuild._runPhases.<fn>.<fn>
package:build_runner_core/src/generate/performance_tracker.dart 184:15  _NoOpBuildPerformanceTracker.trackBuildPhase
package:build_runner_core/src/generate/build_impl.dart 315:47           _SingleBuild._runPhases.<fn>
package:timing/src/timing.dart 222:44                                   NoOpTimeTracker.track
package:build_runner_core/src/generate/build_impl.dart 309:32           _SingleBuild._runPhases
package:build_runner_core/src/logging/logging.dart 25:30                logTimedAsync
package:build_runner_core/src/generate/build_impl.dart 266:26           _SingleBuild._safeBuild.<fn>
dart:async                                                              runZoned
package:build_runner_core/src/generate/build_impl.dart 261:5            _SingleBuild._safeBuild
package:build_runner_core/src/generate/build_impl.dart 208:24           _SingleBuild.run
package:build_runner_core/src/generate/build_impl.dart 94:56            BuildImpl.run
package:build_runner_core/src/generate/build_runner.dart 25:14          BuildRunner.run
package:build_runner                                                    BuildCommand.run
package:args/command_runner.dart 197:27                                 CommandRunner.runCommand
package:args/command_runner.dart 112:25                                 CommandRunner.run.<fn>
dart:async                                                              new Future.sync
package:args/command_runner.dart 112:14                                 CommandRunner.run
package:build_runner                                                    run
.dart_tool\build\entrypoint\build.dart 31:22                            main

[SEVERE] Build:
Failed after 186ms
[+5273 ms] "flutter run" took 5,463ms.
[  +13 ms] pub finished with exit code 1
[   +9 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:14:3)
           #1      _DefaultPub.interactively (package:flutter_tools/src/dart/pub.dart:361:7)
           <asynchronous suspension>
           #2      PackagesForwardCommand.runCommand (package:flutter_tools/src/commands/packages.dart:235:15)
           #3      FlutterCommand.verifyThenRunCommand (package:flutter_tools/src/runner/flutter_command.dart:860:18)
           #4      _rootRunUnary (dart:async/zone.dart:1198:47)
           #5      _CustomZone.runUnary (dart:async/zone.dart:1100:19)
           #6      _FutureListener.handleValue (dart:async/future_impl.dart:143:18)
           #7      Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:696:45)
           #8      Future._propagateToListeners (dart:async/future_impl.dart:725:32)
           #9      Future._completeWithValue (dart:async/future_impl.dart:529:5)
           #10     Future._asyncCompleteWithValue.<anonymous closure> (dart:async/future_impl.dart:567:7)
           #11     _rootRun (dart:async/zone.dart:1190:13)
           #12     _CustomZone.run (dart:async/zone.dart:1093:19)
           #13     _CustomZone.runGuarded (dart:async/zone.dart:997:7)
           #14     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1037:23)
           #15     _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
           #16     _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
           #17     _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:118:13)
           #18     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:169:5)


[ +351 ms] ensureAnalyticsSent: 265ms
[   +3 ms] Running shutdown hooks
[  +13 ms] Shutdown hooks complete
[   +2 ms] exiting with code 1
```
</details>
<details>
<summary>GraphQL schema:</summary>

```graphql
type Query {
    articles(article_type_in: [ArticleType!]): [Article!]
}

type Article {
  id: ID!
  title: String!
  article_type: ArticleType!
}

enum ArticleType {
  NEWS
  TUTORIAL
}
```
</details>
<details>
<summary>GraphQL query:</summary>

```graphql
query BrowseArticles($article_type_in: [ArticleType!]) {
    articles(article_type_in: $article_type_in) {
        id
        title
        article_type
    }
}
```
</details>

**Comment by vasilich6107** [comment]
Awesome bug description! I'll take a look

**Comment by vasilich6107** [comment]
added fix in this branch `uppercase-keyword-fix`

**Comment by kono0514** [comment]
@vasilich6107 
`uppercase-keyword-fix` is working good! Thank you 👍

**Comment by vasilich6107** [comment]
Fixed in v6.7.1-beta.1
